include/libopencm3/stm32/common/rtc_common_l1f024.h

```diff
@@ -426,6 +426,20 @@ specific memorymap.h header before including this header file.*/
 #define RTC_ALRMXSSR_SS_SHIFT     (0)
 #define RTC_ALARXSSR_SS_MASK      (0x7fff)
 
+enum rtc_fmt {
+	RTC_CR_FMT_24H = 0x0,
+	RTC_CR_FMT_12H,
```

Comment:
I'm not entirely sure a struct is necessary here, could change this to the same method as `rtc_time_set_am_notation` and `rtc_time_set_pm_notation` instead of passing an enum to `rtc_set_hour_format`

On a side note, I'm not entirely sure why the `RTC_CR` has a 12/24h format option **and** the `RTC_TR` time register itself has an AM(24h)/PM format option.

Reply:
Yeah, I don't think you need an enum for this.  Keep those two functions perhaps, btu make them set the RTC_CR flag appropriately, and move teh TR settings elsewherew as I suggested.